File: exercism/clojure/raindrops/src/raindrops.clj
(ns raindrops)

(defn convert [val-int] ;; <- arglist goes here
      ;; your code goes here
  (let [multiple-of-3 (zero? (mod val-int 3))
        multiple-of-5 (zero? (mod val-int 5))
        multiple-of-7 (zero? (mod val-int 7))]
    (cond-> ""
      multiple-of-3 (str "Pling")
      multiple-of-5 (str "Plang")
      multiple-of-7 (str "Plong")
      (every? false? [multiple-of-3 multiple-of-5 multiple-of-7]) (str val-int))))

File: exercism/fortran/raindrops/raindrops.f90
module raindrops
   implicit none
contains

   function convert(i)
      integer :: i
      character(20) :: convert
      convert = ' '
      if(mod(i, 3) == 0) then
         convert = trim(convert) // 'Pling'
      endif
      if(mod(i, 5) == 0) then
         convert = trim(convert) // 'Plang'
      endif
      if(mod(i, 7) == 0) then
         convert = trim(convert) // 'Plong'
      endif

      if(len_trim(convert) == 0) then
         write(convert, '(i10)' )  i
         convert = adjustl(convert)
      endif
   end function convert

end module raindrops

